mautrix_instagram/portal.py

```diff
@@ -972,7 +971,21 @@ async def _handle_instagram_media_share(
             format=Format.HTML,
             external_url=external_url,
         )
-        await self._send_message(intent, caption, timestamp=item.timestamp_ms)
+
+        if self.bridge.config["bridge.caption_in_message"]:
+            combined = content
+            combined["filename"] = combined.body
+            caption_body = f"{prefix.body}\n{caption.body}"
+            combined["org.matrix.msc1767.text"] = caption_body
```

Comment:
This (and `org.matrix.msc1767.html` if applicable) should be inside `org.matrix.msc1767.caption`

(it's also fine to just not include that stuff at all)